internal/config/profile.go

```diff
@@ -151,7 +154,7 @@ func Exists(name string) bool {
 }
 
 func newProfile() *Profile {
-	configDir, err := configHome()
+	configDir, err := configHome(ToolName)
 	if err != nil {
 		log.Fatal(err)
 	}
```

Comment:
@gssbzn I am not sure I understood what you proposed. I am already injecting the config dir based on the `ToolName`in `newProfile`. I am not sure what `newAtlasProfile` should do.

Reply:
what you are currently doing is setting the configDir based on the global state of `ToolName` you are injecting to `p` yes but `newProfile` has no concept of injection, new profile should not have an if to decide  what home dir to use
it's better to keep `newProfile` as is, and create a new method, `newAtlasProfile` and none of them have an if based on global state

Reply:
we still have to use the global state of `ToolName` to decide which newProfile function to call, right? Where should I add this logic? I see that we are using `var p = newProfile()` to create a new profile now

Reply:
`var p = newProfile()` was meant to simulate a singleton for convenience, this introduced some problems on testing and why we have slowly moved to inject `config.Default()` into commands instead of using `config.SetX` or `config.GetX` we should move away from those global setters and global state.

Prefer a specific method, ie `config.LoadAtlas`, `config.SaveAtlas` and the only place to probably compromise is `config.Default` to have an `if` to keep it backward compatible